acceptance_test.go

```diff
@@ -0,0 +1,143 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mongo
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/brianvoe/gofakeit"
+	"github.com/conduitio-labs/conduit-connector-mongo/config"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/matryer/is"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+const (
+	// set the directConnection to true in order to avoid the known hostname problem.
+	testURI              = "mongodb://localhost:27017/?directConnection=true"
+	testDB               = "test_acceptance"
+	testCollectionPrefix = "test_acceptance_coll"
+)
+
+type driver struct {
+	sdk.ConfigurableAcceptanceTestDriver
+}
+
+// GenerateRecord overrides the [sdk.ConfigurableAcceptanceTestDriver] GenerateRecord method.
+// It generates a MongoDB-specific payload and a random bson.ObjectID key converted to a string.
+func (d driver) GenerateRecord(t *testing.T, operation sdk.Operation) sdk.Record {
+	t.Helper()
+
+	id := primitive.NewObjectID().String()
+
+	return sdk.Record{
+		Operation: operation,
+		Key: sdk.StructuredData{
+			"_id": id,
+		},
+		Payload: sdk.Change{
+			After: sdk.StructuredData{
+				"_id":        id,
+				"name":       gofakeit.Name(),
+				"email":      gofakeit.Email(),
+				"created_at": time.Now().Format(time.RFC3339),
+			},
+		},
+	}
+}
+
+func TestAcceptance(t *testing.T) {
+	cfg := map[string]string{
+		config.KeyURI: testURI,
+		config.KeyDB:  testDB,
+	}
+
+	sdk.AcceptanceTest(t, driver{
+		sdk.ConfigurableAcceptanceTestDriver{
+			Config: sdk.ConfigurableAcceptanceTestDriverConfig{
+				Connector:         Connector,
+				SourceConfig:      cfg,
+				DestinationConfig: cfg,
+				BeforeTest:        beforeTest(cfg),
+				AfterTest:         afterTest(cfg),
+			},
+		},
+	})
+}
+
+// beforeTest set the config collection field to a unique name prefixed with the testCollectionPrefix.
+func beforeTest(cfg map[string]string) func(*testing.T) {
+	return func(t *testing.T) {
+		t.Helper()
+
+		is := is.New(t)
+
+		// create a test mongo client
+		mongoClient, err := createTestMongoClient(context.Background(), cfg[config.KeyURI])
+		is.NoErr(err)
+		defer func() {
+			err = mongoClient.Disconnect(context.Background())
+			is.NoErr(err)
+		}()
+
+		cfg[config.KeyCollection] = fmt.Sprintf("%s_%d", testCollectionPrefix, time.Now().UnixNano())
+
+		// connect to the test database and create a collection
+		testDatabase := mongoClient.Database(cfg[config.KeyDB])
+		is.NoErr(testDatabase.CreateCollection(context.Background(), cfg[config.KeyCollection]))
```

Comment:
Hm. Not sure.
At least we have to init `mongo.Collection` object (of mongodb driver for Go) to use `Insert`, `Update`, or `Delete` methods of certain collection.

Reply:
Right, but there is no usage of the collection here?

Reply:
I don't mind the function being there, but I don't fully understand how does it make the connector execution more predictable?

Reply:
If the user specifies a db or collection name with an invalid name, the connector returns an error, rather than creating a new database or collection.

Reply:
You're making a great point. A simple typo in a destination config would send all the data to the wrong collection. Thanks for the clarification!